auth/src/main/java/com/firebase/ui/auth/ui/idp/AuthMethodPickerActivity.java

```diff
@@ -99,27 +93,51 @@ protected void onCreate(Bundle savedInstanceState) {
             ImageView logo = findViewById(R.id.logo);
             logo.setImageResource(logoId);
         }
+
+        mHandler.getOperation().observe(this, new Observer<Resource<IdpResponse>>() {
+            @Override
+            public void onChanged(Resource<IdpResponse> resource) {
+                if (resource.getState() == State.LOADING) {
+                    getDialogHolder().showLoadingDialog(R.string.fui_progress_dialog_signing_in);
+                    return;
+                }
+                getDialogHolder().dismissDialog();
+
+                if (resource.isUsed()) { return; }
```

Comment:
Looking at this pattern, I wonder if this would be cleaner:

  * Introduce a new `UsableResource` class
  * Have it have the same three states are we have now, but also a `USED` state

Would that work?  Better or worse?

Reply:
I thought of doing something like that too, but then how do we use the `LOADING` state? Our problem is that actions sometimes need to happen every time the activity restarts and other times only once. So far, the terminal-ish states like `SUCCESS` and `FAILURE` seem to only be processed if the resource isn't used. Maybe we could make the `LOADING` state special somehow?

Reply:
Wouldn't actions that need to be processed every time just not be `Usable` then?